src/main/java/org/broadinstitute/hellbender/utils/read/ReadUtils.java

```diff
@@ -1203,16 +1189,18 @@ public static int getFirstAlignedBaseOffset(final GATKRead read) {
         if (read.isUnmapped()) {
             return -1;
         } else {
+            final List<CigarElement> cigarElements = read.getCigarElements();
+            if (cigarElements.isEmpty()) {
+                throw new IllegalArgumentException("the input read is mapped yet contains no cigar-elements");
+            }
             int result = 0;
-            for (final CigarElement ce : read.getCigarElements()) {
+            for (final CigarElement ce : cigarElements) {
                 final int length = ce.getLength();
-                if (length != 0) { // paranoia test.
-                    final CigarOperator co = ce.getOperator();
-                    if (co.isAlignment()) {
-                        return result;
-                    } else if (co.consumesReadBases()) {
-                        result += length;
-                    }
+                final CigarOperator co = ce.getOperator();
+                if (co.isAlignment()) {
+                    return result;
+                } else if (co.consumesReadBases()) {
+                    result += length;
                 }
             }
             return -1;
```

Comment:
I believe the code should not reach here:

The code would reach this point only if the input read is mapped, but its cigar contains no alignment operation.
In this case, the cigar would be composed of {`H`, `S`, `I`, `D`, `N`, `P`}. This would either
* be inconsistent with other functions behavior where a cigar with only clipping operations would throw, or
* be inconsistent with assertion here that read should not be unmapped (unmapped read is caught earlier); in other words, mapped reads with no alignment operation in its cigar is wrong to me.